contracts/index/IndexToken.sol

```diff
@@ -125,7 +123,10 @@ contract IndexToken is
         _setupRole(DEFAULT_ADMIN_ROLE, addressRegistry.emergencySafeAddress());
         _setupRole(ADMIN_ROLE, addressRegistry.adminSafeAddress());
         _setupRole(EMERGENCY_ROLE, addressRegistry.emergencySafeAddress());
-        _setupRole(CONTRACT_ROLE, addressRegistry.mAptAddress());
+        _setupRole(
```

Comment:
We should also take the opportunity, as we upgrade each component, to remove reliance on the address registry.

The constructor should take the addresses as a required parameter and there should be a permissioned setter for each address. Depending on how many parameters this creates for a constructor, it could make sense to have a struct parameter, such as `SafeAddresses`, a la Code Complete (IIRC).